include/dr/mhp/containers/rows.hpp

```diff
@@ -0,0 +1,238 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "index.hpp"
+
+namespace dr::mhp {
+
+using key_type = index<>;
+
+template <typename T, typename Allocator = std::allocator<T>>
+class distributed_dense_matrix;
+
+template <typename DM> class d_segment;
+// template <typename DM> class d_segment_iterator;
+// template <typename T, typename Allocator = std::allocator<T>> class dm_row;
+
+template <typename T, typename Allocator = std::allocator<T>> class dm_row : public std::span<T> {
+  using dmatrix = distributed_dense_matrix<T>;
+  using dsegment = d_segment<dmatrix>;
+
+public:
+  using iterator = typename std::span<T>::iterator;
+
+  dm_row(){};
+  dm_row(signed long idx, T *ptr, std::size_t size, dsegment *segment, 
+         Allocator allocator = Allocator())
+      : std::span<T>({ptr, size}), index_(idx), data_(ptr), segment_(segment),
+        size_(size), allocator_(allocator){};
```

Comment:
`ptr` and `size` are already in `std::span`
no need to double it in dm_row

Reply:
Done